services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserGroupController.java

```diff
@@ -0,0 +1,265 @@
+@RequiredArgsConstructor
+public class UserGroupController {
+
+  private final GroupRepository groupRepository;
+
+  private final UserGroupsRepository userGroupsRepo;
+
+  @Value("${one.owner.msg}")
+  private String oneOwnerMsg;
+
+  @Value("${user.group.not.found}")
+  private String usrGrpNotFound;
+
+  @Value("${only.grp.owner.access}")
+  private String grpOwnerAccess;
+
+  @PostMapping("{id}/user-groups")
+  public ResponseEntity<Object> createRole(
+    @Valid @RequestBody UserGroup userGroup,
+    @PathVariable("id") UUID id
+  ) {
+    Optional<UserGroup> usrGr;
+    Group svdGroup;
+    Optional<Group> group = groupRepository.findById(id);
+    if (group.isPresent()) {
+      svdGroup = group.get();
+      if (userGroup.getGroup().getId() == null) {
+        userGroup.getGroup().setId(svdGroup.getId());
+      }
+      usrGr =
+        userGroupsRepo.findByGroupIdAndUserTenantId(
+          userGroup.getGroup().getId(),
+          userGroup.getUserTenant().getId()
+        );
+      if (!usrGr.isPresent()) {
+        usrGr = (Optional.ofNullable(userGroupsRepo.save(userGroup)));
+        group.get().setModifiedOn(usrGr.get().getModifiedOn());
+        groupRepository.save(group.get());
+      }
+    } else {
+      throw new ResponseStatusException(
+        HttpStatus.NOT_FOUND,
+        CommonConstants.NO_GRP_PRESENT + id
+      );
+    }
+    return new ResponseEntity<>(usrGr.get(), HttpStatus.CREATED);
+  }
+
+  @Transactional
+  @PatchMapping("{id}/user-groups/{userGroupId}")
+  public ResponseEntity<Object> updateAll(
+    @PathVariable("id") UUID id,
+    @RequestBody UserGroup userGroup,
+    @PathVariable("userGroupId") UUID userGroupId
+  ) {
+    /** fetch value in Group against primary key and also to
+         and to update modifiedOn parameter*/
+    Optional<Group> group = groupRepository.findById(id);
+    if (group.isPresent()) {
+      Count count = Count
+        .builder()
+        .totalCount(userGroupsRepo.countByGroupIdAndIsOwner(id, true))
+        .build();
+      if (count.getTotalCount() == 1) {
+        Optional<UserGroup> usrGrp =
+          userGroupsRepo.findByGroupIdAndIdAndIsOwner(id, userGroupId, true);
+
+        if (usrGrp.isPresent() && !usrGrp.get().isOwner()) {
+          throw new ResponseStatusException(HttpStatus.FORBIDDEN, oneOwnerMsg);
+        }
+      }
+      if (userGroup.getId() == null) {
+        userGroup.setId(userGroupId);
+      }
+      Optional<UserGroup> targetUserGroupOptional = userGroupsRepo.findById(
+        userGroupId
+      );
+      UserGroup tarUserGroup = new UserGroup();
+      if (targetUserGroupOptional.isPresent()) {
+        tarUserGroup = targetUserGroupOptional.get();
+      }
+      BeanUtils.copyProperties(
+        userGroup,
+        tarUserGroup,
+        CommonUtils.getNullPropertyNames(userGroup)
+      );
+      userGroupsRepo.save(tarUserGroup);
+
+      group.get().setModifiedOn(LocalDateTime.now());
+      groupRepository.save(group.get());
```

Comment:
since we are not updating any properties of group, we are just updating/ creating the user group and only groups relation userGroups is getting affected, does this required @mlakdawala-sf sir